lua/plenary/async/async.lua

```diff
@@ -104,9 +101,17 @@ M.run = function(async_function, callback)
   end
 end
 
----this needs to be fixed
-M.void = function(async_fun)
-  return co.wrap(async_fun)
+---Use this to create a function which executes in an async context but
+---called from a non-async context. Inherently this cannot return anything
+---since it is non-blocking
+---@param func function
+M.void = function(func)
+  return function(...)
+    local args = {...}
+    execute(function()
+      func(unpack(args))
```

Comment:
I think we can add a `vararg` to `execute` so we can pass in the initial values without having to pack and unpack

Reply:
👍